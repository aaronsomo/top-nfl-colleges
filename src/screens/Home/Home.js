import React from 'react';
import { SafeAreaView, StyleSheet, Text, Button, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';


const Home = ({ navigation }) => {
    return (
        <SafeAreaView style={styles.container}>
            <Text style={styles.header}>NFL Talent Scout</Text>
            <View style={styles.scoutButtonContainer}>
                <TouchableOpacity
                    style={styles.disabledScoutBtn}
                    onPress={() => navigation.navigate('SelectTeams')}
                >
                    <Text style={styles.text}>
                        Select Teams >>
                    </Text>
                </TouchableOpacity>
            </View>
        </SafeAreaView>
    )
}

const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: '#3898EC',
      alignItems: 'center',
      justifyContent: 'center',
    },
    header: {
        height: 60,
        width: 400,
        fontSize: 30,
        fontWeight: '600',
        textAlign: 'center',
        color: '#fff',
    },
    scoutButtonContainer: {
        height: 60,
        width: 120,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#3898EC',
        borderRadius: 10,
        margin: 15,
    },
    disabledScoutBtn: {
        height: 60,
        width: 130,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#595faa',
        borderRadius: 10,
        margin: 15,
    },
    text: {
        fontSize: 18,
        fontWeight: '500',
        color: '#fff',
        textAlign: 'center',
    },
  });

export default Home;