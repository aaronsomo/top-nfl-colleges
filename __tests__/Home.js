import 'react-native';
import React from 'react';
import Home from '../src/screens/Home/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

describe('<Home />', () => {
    it('renders correctly', () => {
        renderer.create(<Home />);
    });

    it('should match snapshot', () => {
        const TestHome = render(<Home />).toJSON();
        expect(TestHome).toMatchSnapshot();
    });

    it('navigates to SelectTeams screen', () => {
        const navigate = jest.fn();
        const { getByText } = render(<Home navigation={{ navigate }} />);
        fireEvent.press(getByText('Select Teams >>'));
        expect(navigate).toHaveBeenCalledWith('SelectTeams');
    });

});