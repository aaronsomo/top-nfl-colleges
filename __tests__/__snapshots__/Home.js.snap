// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3898EC",
      "display": "flex",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "#3898EC",
        "borderRadius": 10,
        "borderWidth": 2,
        "display": "flex",
        "height": 400,
        "margin": 10,
        "width": 400,
      }
    }
  >
    <Image
      source={
        Object {
          "canInstrument": true,
          "createTransformer": [Function],
          "getCacheKey": [Function],
          "process": [Function],
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": "100%",
          "opacity": 1,
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 18,
        "fontWeight": "500",
        "textAlign": "center",
      }
    }
  >
    Scout Top NFL Colleges by Teams
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3898EC",
        "borderRadius": 10,
        "height": 60,
        "justifyContent": "center",
        "margin": 15,
        "width": 120,
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#595faa",
            "borderRadius": 10,
            "height": 60,
            "justifyContent": "center",
            "margin": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 18,
              "fontWeight": "500",
              "textAlign": "center",
            }
          }
        >
          Select Teams &gt;&gt;
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</RCTSafeAreaView>
`;
